core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeAstBuilder.scala

```diff
@@ -56,31 +56,130 @@ class SnowflakeAstBuilder extends SnowflakeParserBaseVisitor[TreeNode] {
+      } else if (joinType.outer_join().LEFT() != null) {
+        ir.LeftOuterJoin
+      } else if (joinType.outer_join().RIGHT() != null) {
+        ir.RightOuterJoin
+      } else if (joinType.outer_join().FULL() != null) {
+        ir.FullOuterJoin
+      } else {
+        ir.UnspecifiedJoin
+      }
+    }
+
+    def buildJoin(left: ir.Relation, right: SnowflakeParser.Join_clauseContext): ir.Join = {
+
+      ir.Join(
+        left,
+        right.object_ref().accept(this),
+        None,
+        translateJoinType(right.join_type()),
+        Seq(),
+        ir.JoinDataType(is_left_struct = false, is_right_struct = false))
+    }
+    val left = ctx.object_ref().accept(this)
+    ctx.join_clause().asScala.foldLeft(left)(buildJoin)
+  }
+
 }
-class SnowflakeExpressionBuilder extends SnowflakeParserBaseVisitor[Expression] {
 
-  override def visitSelect_list_elem(ctx: SnowflakeParser.Select_list_elemContext): Expression = {
+class SnowflakeExpressionBuilder extends SnowflakeParserBaseVisitor[ir.Expression] {
+
+  override def visitSelect_list_elem(ctx: SnowflakeParser.Select_list_elemContext): ir.Expression = {
     val column = ctx.column_elem().accept(this)
     if (ctx.as_alias() != null) {
       ctx.as_alias().accept(this) match {
-        case Alias(_, name, metadata) => Alias(column, name, metadata)
+        case ir.Alias(_, name, metadata) => ir.Alias(column, name, metadata)
         case _ => null
       }
     } else {
       column
     }
   }
-  override def visitColumn_name(ctx: SnowflakeParser.Column_nameContext): Expression = {
-    Column(ctx.id_(0).getText)
+  override def visitColumn_name(ctx: SnowflakeParser.Column_nameContext): ir.Expression = {
+    ir.Column(ctx.id_(0).getText)
   }
 
-  override def visitAs_alias(ctx: SnowflakeParser.As_aliasContext): Expression = {
+  override def visitAs_alias(ctx: SnowflakeParser.As_aliasContext): ir.Expression = {
     val alias = ctx.alias().id_().getText
-    Alias(null, Seq(alias), None)
+    ir.Alias(null, Seq(alias), None)
+  }
+
+  override def visitPrimitive_expression(ctx: Primitive_expressionContext): Expression = {
+    val columnName = ctx.id_(0).getText
+    ir.Column(columnName)
   }
 }
+
+class SnowflakePredicateBuilder extends SnowflakeParserBaseVisitor[ir.Predicate] {
+  override def visitExpr(ctx: ExprContext): Predicate = {
+
+    def buildComparison(left: Expression, right: Expression, op: Comparison_operatorContext): Predicate = {
+      if (op.EQ() != null) {
+        ir.Equals(left, right)
+      } else if (op.NE() != null || op.LTGT() != null) {
+        ir.NotEquals(left, right)
+      } else if (op.GT() != null) {
+        ir.GreaterThan(left, right)
+      } else if (op.LT() != null) {
+        ir.LesserThan(left, right)
+      } else if (op.GE() != null) {
+        ir.GreaterThanOrEqual(left, right)
+      } else if (op.LE() != null) {
+        ir.LesserThanOrEqual(left, right)
+      } else {
+        // TODO: better error management
```

Comment:
TODO: @nfx : automatically create issues from every `TODO: ` in the code

Reply:
TODO inception :)